source/Src/Unity-CoreClr/UnityContainer.cs

```diff
@@ -478,8 +480,12 @@ public void Dispose()
         /// method, false if being called from a finalizer.</param>
         protected virtual void Dispose(bool disposing)
         {
-            if (disposing)
+            if (disposing && !inDispose)
             {
+                // Avoid infinite loop when someone
+                //  registers something which would end up 
+                //  disposing this container (e.g. container.RegisterInsance(container))
+                inDispose = true;
                 if (lifetimeContainer != null)
                 {
                     lifetimeContainer.Dispose();
```

Comment:
Ah that's a good idea, yes it's in `lifetimeContainer` that the SO happens. Will do that (the extra private field did annoy me too).